src/components/v5/common/CompletedAction/partials/SetUserRoles/SetUserRoles.tsx

```diff
@@ -81,8 +97,22 @@ const SetUserRoles = ({ action }: Props) => {
     fromDomain,
     isMotion,
     annotation,
+    blockNumber,
+    colonyAddress,
   } = action;
-  const userColonyRoles = transformActionRolesToColonyRoles(roles);
+
+  const { data: historicRoles } = useGetColonyHistoricRoleRolesQuery({
+    variables: {
+      id: `${colonyAddress}_${fromDomain?.nativeId}_${recipientAddress}_${blockNumber}_roles`,
+    },
+    fetchPolicy: 'cache-and-network',
+  });
+
+  const userColonyRoles = transformActionRolesToColonyRoles(
+    roles,
+    historicRoles?.getColonyHistoricRole,
```

Comment:
Since this function expects two arguments, and the second one here is conditional, it may be a good idea to add a fallback to an empty object or something?

Reply:
Ah, ignore me, I just looked into it and the type includes null or undefined :)

Reply:
You may just need to consider that inside the function you call `const combinedRoles = { ...historicRoles };`which could be unhappy if historicRoles is indeed `null`.

Reply:
Thanks for approving @davecreaser ! Yep that should be fine because in that scenario, `combinedRoles` will just equal to an empty object `{ }` then it will be populated with the new roles that were toggled -- the function will still pretty much behave like it did before 👌